tests/integ/modin/series/test_apply_and_map.py

```diff
@@ -0,0 +1,728 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import datetime
+import math
+from typing import Callable
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+from pandas.testing import assert_series_equal
+from pytest import param
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from snowflake.snowpark._internal.utils import (
+    TempObjectType,
+    random_name_for_temp_object,
+)
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from snowflake.snowpark.functions import udf
+from snowflake.snowpark.types import DoubleType, StringType, VariantType
+from tests.integ.modin.utils import (
+    PANDAS_VERSION_PREDICATE,
+    ColumnSchema,
+    assert_snowpark_pandas_equal_to_pandas,
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck,
+    create_snow_df_with_table_and_data,
+    create_test_series,
+    eval_snowpark_pandas_result,
+)
+from tests.integ.utils.sql_counter import SqlCounter, sql_count_checker
+
+pytestmark = pytest.mark.skipif(
+    PANDAS_VERSION_PREDICATE,
+    reason="SNOW-1739034: tests with UDFs/sprocs cannot run without pandas 2.2.3 in Snowflake anaconda",
+)
+
+BASIC_DATA_FUNC_RETURN_TYPE_MAP = [
+    ([1, 2, 3, None], lambda x: x + 1, "int"),
+    param(
+        [1, 2, 3, None],
+        lambda x: native_pd.Timedelta(x),
+        "native_pd.Timedelta",
+        id="return_timedelta_scalar",
+        marks=pytest.mark.xfail(
+            strict=True, raises=AssertionError, reason="SNOW-1619940"
+        ),
+    ),
+    (["s", "n", "o", "w"], lambda x: x * 2, "str"),
+    ([1.0, 1.5, 2.0], math.exp, "float"),
+    ([True, True, False], int, "int"),
+    ([bytes("snow", "utf-8"), bytes("flake", "utf-8")], lambda x: x.decode(), "str"),
+    ([1, 2, 3], lambda x: [x, x + 1], "list"),
+    ([1, 2, 3], lambda x: {str(x): x * 8}, "dict"),
+    ([None, 1], lambda x: bool(np.isnan(x)), "bool"),
+    ([np.nan, None, 2.1], lambda x: x, "float"),
+    param(
+        [native_pd.Timedelta(1), native_pd.Timedelta(2)],
+        lambda x: x.value,
+        "int",
+        id="apply_on_frame_with_timedelta_data_columns_returns_int",
+        marks=pytest.mark.xfail(strict=True, raises=NotImplementedError),
+    ),
+]
+
+# TODO SNOW-876999: Test return date/time/timestamp type when
```

Comment:
This is marked as done. Can we either remove the todo (if truly resolved) or create a new ticket to capture the remaining work to be done?

Reply:
There's follow-up work required. I filed a new issue, SNOW-1798212, and I linked to it here.